### PR TITLE
Implement VK_EXT_device_generated_commands.

### PR DESCRIPTION
**Pull request by HansKristian-Work**
Supersedes https://github.com/HansKristian-Work/vkd3d-proton/pull/2135.
Continues where Mike left off since he's too busy with other things.

### CODE REVIEW

**Comment by HansKristian-Work** [comment]
Passes on https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/31383.

NV driver seems to have an odd bug where it computes the required size completely wrong (several GB in some cases).

File: libs/vkd3d/command.c [hunk 1]
```diff
@@ -19534,7 +19712,58 @@ static HRESULT d3d12_command_signature_allocate_stream_memory_for_list(
...
+    VkGeneratedCommandsPipelineInfoEXT pipeline_info;
+    VkMemoryRequirements2 memory_info;
+    uint32_t alignment;
+
+    memory_info.sType = VK_STRUCTURE_TYPE_MEMORY_REQUIREMENTS_2;
+    memory_info.pNext = NULL;
+
+    pipeline_info.sType = VK_STRUCTURE_TYPE_GENERATED_COMMANDS_PIPELINE_INFO_EXT;
+    pipeline_info.pNext = NULL;
+    pipeline_info.pipeline = render_pipeline;
```
Comment (doitsujin) [inline libs/vkd3d/command.c:19732]: can we just memset these for consistency too?

**Review by doitsujin — CHANGES_REQUESTED** [review]
Nit.

Insane amount of boilerplate and duplication though, I hope we'll be able to drop the NV path eventually.

**Comment by HansKristian-Work** [comment]
> I hope we'll be able to drop the NV path eventually.

Yes, once EXT lands in stableOS, we can nuke NV path.

File: libs/vkd3d/command.c (hunk 1 above)
Comment (HansKristian-Work) [reply libs/vkd3d/command.c:19732]: fixed.

File: libs/vkd3d/device.c [hunk 2]
```diff
@@ -2578,6 +2589,32 @@ static HRESULT vkd3d_init_device_caps(struct d3d12_device *device,
...
+
+    {
+        const VkShaderStageFlags supported_stages =
+                physical_device_info->device_generated_commands_properties_ext.supportedIndirectCommandsShaderStages;
+        VkShaderStageFlags required_stages = VK_SHADER_STAGE_ALL_GRAPHICS | VK_SHADER_STAGE_COMPUTE_BIT;
+
+        if (physical_device_info->mesh_shader_features.meshShader)
+            required_stages |= VK_SHADER_STAGE_MESH_BIT_EXT;
+        if (physical_device_info->mesh_shader_features.taskShader)
+            required_stages |= VK_SHADER_STAGE_TASK_BIT_EXT;
```
[the remark this answers is not in the document]
Comment (HansKristian-Work) [reply libs/vkd3d/device.c]: fixed.

**Review by doitsujin — APPROVED** [review]

**Pull request merged**